coderd/database/queries/workspaceagentportshare.sql

```diff
@@ -9,3 +9,14 @@ UPDATE workspace_agent_port_share SET share_level = $1 WHERE workspace_id = $2 A
 
 -- name: DeleteWorkspaceAgentPortShare :exec
 DELETE FROM workspace_agent_port_share WHERE workspace_id = $1 AND agent_name = $2 AND port = $3;
+
+-- name: RestrictWorkspaceAgentPortSharesByTemplate :exec
+UPDATE workspace_agent_port_share
+SET share_level = $1
+WHERE workspace_id IN (
+	SELECT id
+	FROM workspaces
+	WHERE template_id = $2
+) AND share_level > $1;
```

Comment:
When you change this to enums this probably won't work anymore, and instead you'll probably want to pass in a list of "allowed share levels" and delete if it's not in that list. kinda like `share_level NOT IN $1::share_level[]`

Reply:
Yeah gonna bring this in after the main PR is merged, this is helpful advice.